hibernate/cache-commons/src/test/java/org/infinispan/test/hibernate/cache/commons/functional/ReadOnlyTest.java

```diff
@@ -91,6 +94,44 @@ public void testInsertClearCacheDeleteEntity() throws Exception {
 		});
 	}
 
+   @Test
+   @TestForIssue(jiraKey = "ISPN-9369")
+   public void testCacheQueryResultsOnRollback() throws Exception {
```

Comment:
I believe that the request in JIRA requests to expose the query results when the transaction is in-flight (as it's never committed), not just on rollback.